statistics/feedback.go

```diff
@@ -336,14 +320,24 @@ func (b *BucketFeedback) splitBucket(newBktNum int, totalCount float64, count fl
 			bounds[i] = bounds[i-1]
 			continue
 		}
-		bkts = append(bkts, bucket{lower: &bounds[i-1], upper: &bounds[i], count: newCount, repeat: 0})
+		bkts = append(bkts, bucket{lower: &bounds[i-1], upper: bounds[i].Copy(), count: newCount, repeat: 0})
+		// To guarantee that each bucket's range will not overlap.
+		if bounds[i].Kind() == types.KindBytes {
+			bounds[i].SetBytes(kv.Key(bounds[i].GetBytes()).PrefixNext())
+		} else if bounds[i].Kind() == types.KindInt64 {
+			bounds[i].SetInt64(bounds[i].GetInt64() + 1)
```

Comment:
Can it overflow max int64?

Reply:
No, because the `bounds` is sorted and there is no same value, so this `bounds` must be less than max int64.